src/runtime/substitution.cc

```diff
@@ -2192,6 +2318,112 @@ std::unique_ptr<Graph> GraphSearchHelper::base_optimize(
   return std::unique_ptr<Graph>(best_graph);
 }
 
+/**
+ * @brief Experimental. Smallest sub-problem of Unity's DP search algorithm with
+ * memory consideration.
+ *
+ * @param r_graph Graph to be optimized
+ * @param simplification_settings Settings to simplify the PCG
+ * @return std::unique_ptr<Graph> Optimized PCG
+ */
+std::unique_ptr<Graph> GraphSearchHelper::base_optimize_with_memory(
```

Comment:
Any way to break up this function a bit?

Reply:
Let's plan the refactor in the future.

Reply:
In that case let's get an issue created and a link to the issue added here and in the code

Reply:
https://github.com/flexflow/FlexFlow/issues/620

Reply:
Can you add this todo specifically in the issue (and the others in this PR as well) just so they don't get forgotten?

Reply:
If you don't want that issue to get too large I'd be fine with multiple issues (in fact it might be preferred)